### PR TITLE
Added LanItems summary

### PR DESCRIPTION
**Pull request by teclator**
This is just the firs request for comments, I could make it simpler if needed or if we consider it as over-engineering.

I decided to use the LanItems.summary but permitting to used different types of summaries being the Default based on Summary class as it is in others networks modules like:

https://github.com/yast/yast-network/blob/master/src/modules/Host.rb#L263
https://github.com/yast/yast-network/blob/master/src/modules/DNS.rb#L532
https://github.com/yast/yast-network/blob/master/src/modules/DNS.rb#L532

So for CASP one click dialog we could use:

 `Yast::LanItems.summary("one_line")`
or
`Yast::LanItemsSummary.new.one_line`

> === OLD ===
> 
>  `Yast::LanItems.summary(:type => "one_line")`
> or
> `Yast::LanItemsSummary.new(:type => "one_line").summary`
> or
>   `Yast::LanItemsSummary::OneLine.new.summary`
> 

And in case of a proposal in the future just `Yast::LanItems.summary`

Here screenshots with the current CASP one dialog overview: yast/yast-installation#487

### With one interface configured 
networkoneinterface

### With multiple interfaces and different bootproto
networkmultipleinterfaces

### CODE REVIEW

**Comment by imobachgs** [comment]
I'm fine with the general approach (having a new class that takes care of generating the summary), but I would simplify it a little bit:

* Using classes for each type of summary looks overkill to me (after all, logic is not that complex and you are not maintaining state). Maybe I would implement different methods and that's all.
* I would avoid using a generic `@options` hash. I rather prefer to use real parameters (using keywords, for example). If we're going to have a lot of options, then `@options` would be a reasonable approach.

Just my opinion :)

**Comment by mchf** [comment]
for the first view it seems to be a bit over engineered, I'll give it even second look ;-)

**Comment by teclator** [comment]
I tried to make it general and as much open as possible. For example with the **OneLine** type currently we only show on interface or multiple interfaces, but maybe we could change the number of the items to show per line being an option to pass, but don't like to use as a specific option in LanItem, but as commented.

The current types are mostly like presenters, I could by now instead of create separate class just call LanItemsSummary.send(@summary_type) and move the logic to methods as @imobachgs has suggested.

File: src/lib/network/lan_items_summary.rb [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
```
Comment (mvidner) [inline src/lib/network/lan_items_summary.rb]: The license is trunca

Comment (teclator) [inline src/lib/network/lan_items_summary.rb]: Fixed

**Comment by teclator** [comment]
@mchf @mvidner ok, as no updates I will modify current PR based on @imobachgs review.

File: test/lan_items_summary_test.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      { "IPADDR" => "1.2.3.4", "NETMASK" => "255.255.255.0" }
+    ].freeze
+  end
+
+  let(:items) do
+    {
+      0 => { "ifcfg" => "eth0" },
+      1 => { "ifcfg" => "eth1" },
+      2 => { "ifcfg" => "br0" }
+    }
```
Comment (imobachgs) [inline test/lan_items_summary_test.rb]: .freeze?

File: test/lan_items_summary_test.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  before do
+    allow(Yast::LanItems).to receive(:Items).and_return(items)
+    allow(Yast::LanItems).to receive(:IsItemConfigured).and_return(true)
+    dhcp_maps.each_with_index do |item, index|
+      allow(Yast::LanItems).to receive(:GetDeviceMap).with(index).and_return(item)
+    end
+  end
+
+  describe "#default" do
+    it "retuns a Richtext summary of the configured interfaces" do
```
Comment (imobachgs) [inline test/lan_items_summary_test.rb]: returns

File: test/lan_items_summary_test.rb [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  end
+
+  describe "#one_line" do
+    it "returns a plain text summary of the configured interfaces in one line" do
+      expect(subject.one_line).to eql "Multiple Interfaces"
+    end
+
+    context "when there are no configured interfaces" do
+      let(:items) { {} }
+      it "returns 'Not configured'" do
```
Comment (imobachgs) [inline test/lan_items_summary_test.rb]: Above you're referencing to `Yast::Summary.NotConfigured`. Please, keep consistency.

Comment (teclator) [inline test/lan_items_summary_test.rb]: Yes, but Summary.NotConfigured returns richtext 

```Yast::Summary.NotConfigured => <ul><li>Not configured yet.</li></ul>```

Comment (imobachgs) [inline test/lan_items_summary_test.rb]:
```ruby
Yast.import "Summary"
Yast::Summary.NotConfigured #=> "Not configured yet."
```

File: src/lib/network/lan_items_summary.rb [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+      output << protocols.first if protocols.uniq.size == 1
+
+      case configured.size
+      when 0
+        return _("Not configured")
+      when 1
+        output << configured.join(", ")
+      else
+        output << "Multiple Interfaces"
```
Comment (kobliha) [inline src/lib/network/lan_items_summary.rb]: PLS mark text for translators

Comment (teclator) [inline src/lib/network/lan_items_summary.rb]: Updated!

File: test/lan_items_summary_test.rb (hunk 4 above)
Comment (teclator) [inline test/lan_items_summary_test.rb]: Sorry, I was thinking in the previous output that was based on Yast::Summary.DevicesList([]), updated!, thnx for check it.

File: test/lan_items_summary_test.rb [hunk 6]
```diff
@@ -19,9 +19,10 @@
       0 => { "ifcfg" => "eth0" },
       1 => { "ifcfg" => "eth1" },
       2 => { "ifcfg" => "br0" }
-    }
+    }.freeze
   end
   let(:no_interfaces) { _("Not configured") }
+  let(:multiple_interfaces) { _("Multiple Interfaces") }
```
Comment (imobachgs) [inline test/lan_items_summary_test.rb]: Maybe these ones are just constants.

**Pull request closed without merging**